### DISCUSSION TITLE
Delete remote branch together with local branch

### DISCUSSION THREAD

**Discussion by EpsilonD3lta** [gitextensions/gitextensions#9516] [Q&A]
## Feature description
Allow to delete local and remote branch in one action. I would like to have a checkbox "Delete remote branch [name]" under Force delete checkbox
![image](https://user-images.githubusercontent.com/30054007/129883504-1cb99207-14e2-470b-a109-69ed5b51688b.png)

**Comment by vbjay** [comment]
That's by design.

**Reply by RussKie** [reply]
Deleting a remote branch is a potential data-loss scenario. This is why we have a separate delete dialog for remote branches with an explicit confirmation.

There several ways you can tackle this:
1. You can probably add a script that runs on `AfterPush` to delete a local branch. 
2. If 1. turns our too tricky, you can create your own script to delete both local and remote branch without the dialogs we provide.
    ![image](https://user-images.githubusercontent.com/4403806/129906884-2b6685e4-cdda-4393-998b-5934c315c0d7.png)

**Reply by EpsilonD3lta** [reply]
I am well aware that it's by design - my issue is not a bug, but feature request. My argument is that the design is inefficient, spending time on something that could be automated.

Potential data loss is always present, it's the probability of it happening that is important. A checkbox that the user has to tick seems to be safe enough for me. But I understand that you might have a different opinion.

**Comment by vbjay** [comment]
We have a plugin you can look at.

**Reply by EpsilonD3lta** [reply]
What is the name of the plugin

**Comment by vbjay** [comment]
Look at obsolete branches. Kinda figured you'd spot it by clicking plugins
menu.

**Comment by vbjay** [comment]
Between that plugin abd the push multiple branches tab in push dialog it's
not hard.

**Comment by EpsilonD3lta** [comment]
Yeah I noticed that, but disregarded it because of the word obsolete as my post is not necessarily about obsolete branches. But you are right, probably any branch that is about to be deleted is in reality obsolete.  And then there is a plugin manager, I thought it might be some package on nuget.org
Anyway, it's a bit clumsy to use for just one branch deletion, it seems to me it is better as general cleanup tool. I will stick to the console for this particular task then.

**Comment by pmiossec** [comment]
I would like also to have this feature.

But for me it makes more sense to delete the local branch when deleting the remote branch.

That's why I did the minimal change to do that and using it since months and find it very convenient...

It lays in this branch (a commit and a fixup) :


https://github.com/pmiossec/gitextensions/commits/rejected_but_useful

**Comment by vbjay** [comment]
Frankly. I think it is foolish. Having that second oh did I mean to delete
that branch chance is a lot better than mass deleting.

**Comment by vbjay** [comment]
Using the obsolete branches to manage local and remove the local branches
first or using delete branches dialog.  Then just push checking delete in
multiple branches tab of push dialog.
